gammapy/maps/tests/test_hpxmap.py

```diff
@@ -198,6 +198,22 @@ def test_hpxmap_crop(nside, nested, coordsys, region, axes):
     m.crop(1)
 
 
+@pytest.mark.parametrize(('nside', 'nested', 'coordsys', 'region', 'axes'),
+                         hpx_test_partialsky_geoms)
+def test_hpxmap_upsample(nside, nested, coordsys, region, axes):
+    m = HpxNDMap(HpxGeom(nside=nside, nest=nested,
+                         coordsys=coordsys, region=region, axes=axes))
+    m.upsample(2)
```

Comment:
It's often hard to add good, useful asserts in parametrised tests.
So for that one could pick a well-chosen simple test case and just put a second test that runs it and then established interpolation / extrapolation behaviour.

Reply:
I've now rewritten the downsample/upsample methods and added some more thorough tests.  At the moment I still don't test the interpolation/extrapolation for `pad` since there's no way to support these modes in healpy.